react-hot-button/my-app/src/App.tsx

```diff
@@ -0,0 +1,41 @@
+import './hot-button.css';
+import './App.css';
+import { HotButton } from './hot-button';
+import { ClickCount } from './click-count';
+import { useState } from 'react';
+
+const buttonText = 'Hot Button';
+let className = 'purple';
```

Comment:
You put this in a global variable, just like Johan did. Here's what I told him:

You are setting className in the handleClick function, so its value gets lost when handleClick returns. It was clever of you to solve this by making className a global variable (by putting it outside of App). However, that's considered bad practice because it can introduce bugs.

For example, suppose you had 2 Hot Buttons on the same page. They would both share the same value of className, so they would be displaying the incorrect color.

There are two ways to solve this. One, you could make className be state. That way it retains its value across renders. The other is to do this computation outside of handleClick, just right inside the App function.

If you had done layout followed by state followed by events, you would have found this problem.